selfservice/strategy/link/test/persistence.go

```diff
@@ -117,44 +112,44 @@ func TestPersister(ctx context.Context, conf *config.Config, p interface {
 				i.VerifiableAddresses = append(i.VerifiableAddresses, *address)
 
 				require.NoError(t, p.CreateIdentity(ctx, &i))
-				return &link.VerificationToken{
+				return &f, &link.VerificationToken{
 					Token:             x.NewUUID().String(),
-					FlowID:            uuid.NullUUID{UUID: req.ID, Valid: true},
+					FlowID:            f.ID,
 					VerifiableAddress: &i.VerifiableAddresses[0],
 					ExpiresAt:         time.Now(),
 					IssuedAt:          time.Now(),
 				}
 			}
 
 			t.Run("case=should error when the verification token does not exist", func(t *testing.T) {
-				_, err := p.UseVerificationToken(ctx, "i-do-not-exist")
+				_, err := p.UseVerificationToken(ctx, x.NewUUID(), "i-do-not-exist")
 				require.Error(t, err)
 			})
 
 			t.Run("case=should create a new verification token", func(t *testing.T) {
-				token := newVerificationToken(t, "foo-user@ory.sh")
+				_, token := newVerificationToken(t, "foo-user@ory.sh")
 				require.NoError(t, p.CreateVerificationToken(ctx, token))
 			})
 
 			t.Run("case=should create a verification token and use it", func(t *testing.T) {
```

Comment:
Please add a test which wants to use a verification token that exists, but with the incorrect flow ID. That should return an error!

Reply:
Done.